src/ethereum_spec_tools/forks.py

```diff
@@ -57,12 +58,69 @@ def discover(cls: Type[H]) -> List[H]:
 
             forks.append(cls(mod))
 
-        forks.sort(key=lambda fork: fork.block)
+        # Timestamps are bigger than block numbers, so this always works.
+        forks.sort(
+            key=lambda fork: fork.block
+            if hasattr(fork, "block")
+            else fork.timestamp
+        )
         if new_package:
             forks.append(new_package)
 
         return forks
 
+    @classmethod
+    def load(cls: Type[H], config_dict: Dict[int, str]) -> List[H]:
+        """
+        Load the forks from a config dict specifying fork blocks and
+        timestamps.
+        """
+        config = sorted(config_dict.items(), key=lambda x: x[0])
+
+        forks = []
+
+        for (block_or_time, name) in config:
+            mod = importlib.import_module("ethereum." + name)
+
+            if block_or_time < 1_000_000_000:
+                mod.MAINNET_FORK_BLOCK = block_or_time  # type: ignore
```

Comment:
Is the `type: ignore` comment because `block_or_time` is the wrong type? You should be able to do an `assert isinstance(block_or_time, U64)` or similar.

Reply:
The issue is that mypy has no way to know whether `mod` has a `MAINNET_FORK_BLOCK` attribute. The typecheck failures being suppressed are:
```
src/ethereum_spec_tools/forks.py:148: error: Module has no attribute "MAINNET_FORK_BLOCK"
src/ethereum_spec_tools/forks.py:152: error: Module has no attribute "MAINNET_FORK_TIMESTAMP"
```